pkg/operation/botanist/component/istio/istiod.go

```diff
@@ -17,22 +17,35 @@ package istio
 import (
 	"context"
 	"path/filepath"
+	"time"
 
-	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/chartrenderer"
+	"github.com/gardener/gardener/pkg/features"
+	gardenletfeatures "github.com/gardener/gardener/pkg/gardenlet/features"
 	"github.com/gardener/gardener/pkg/operation/botanist/component"
 	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
 
+	networkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	// ManagedResourceIstio is the name of the ManagedResource containing the resource specifications.
+	ManagedResourceIstio = "istio"
+)
+
 type istiod struct {
-	namespace string
-	values    *IstiodValues
-	kubernetes.ChartApplier
-	chartPath string
-	client    crclient.Client
+	client                    crclient.Client
+	chartRenderer             chartrenderer.Interface
+	namespace                 string
+	values                    *IstiodValues
+	chartPath                 string
+	istioIngressGatewayValues []*IngressGateway
+	istioProxyProtocolValues  []*IstioProxyProtocol
```

Comment:
same question with pointers

Reply:
With array, pointer seems to me a better option.

Reply:
Why?

Reply:
The same optimization reason will change it.